lib/src/ui/barcode_scanner_widget.dart

```diff
@@ -137,6 +171,39 @@ class _BarcodeScannerWidgetState extends State<BarcodeScannerWidget> {
     }
   }
 
+  Future<bool> _onSupportsSwitchingCameraListener() async {
+    for (final value in _configuredScanners) {
+      if (await value?.controller.supportsSwitchingCamera ?? false) return true;
+    }
+    return false;
+  }
+
+  bool _onTorchStateListener() => _configuredScanners
+      .any((element) => element?.controller.torchState ?? false);
+
+  Future<void> _onToggleCameraListener() async {
+    for (final value in _configuredScanners) {
+      if (await value?.controller.supportsSwitchingCamera ?? false) {
+        await value?.controller.toggleCamera();
+        return;
+      }
+    }
+  }
+
+  bool _onSupportScannerListener<T extends BarcodeScanner>() {
+    return _configuredScanners
+        .whereType<T>()
+        .any((element) => element.controller.isSupported);
+  }
+
+  Future<void> _onToggleTorchListener() async {
+    for (final value in _configuredScanners) {
+      final state = value?.controller.torchState;
```

Comment:
move value to a nonNullableValue because right now you need to do 3 if checks instead of just 1 with a local variable